src/Extension/SandboxExtension.php

```diff
@@ -50,16 +53,21 @@ public function disableSandbox()
         $this->sandboxed = false;
     }
 
-    public function isSandboxed()
+    public function isSandboxed($source = null)
     {
-        return $this->sandboxedGlobally || $this->sandboxed;
+        return $this->sandboxedGlobally || $this->sandboxed || $this->isSourceSandboxed($source);
     }
 
     public function isSandboxedGlobally()
     {
         return $this->sandboxedGlobally;
     }
 
+    public function isSourceSandboxed($source)
```

Comment:
This should have parameter and return type (for new methods, there is no backward compatibility issue preventing us to add it immediately rather than in the next major version)

Reply:
Done